lib/oli/delivery/sections.ex

```diff
@@ -561,6 +561,58 @@ defmodule Oli.Delivery.Sections do
     |> Repo.all()
   end
 
+  @doc """
+  Gets all sections that use a particular project when their 'end_date' attribute is not nil and is later than the current date.
+
+  ## Examples
+      iex> get_active_sections_by_project(project, now_time)
+      [%Section{}, ...]
+
+      iex> get_active_sections_by_project(invalid_project, now_time)
+      []
+  """
+  def get_active_sections_by_project(project_id) do
+    today = DateTime.utc_now()
+
+    Repo.all(
+      from(
+        section in Section,
+        join: spp in SectionsProjectsPublications,
+        on: spp.section_id == section.id,
+        where:
+          spp.project_id == ^project_id and
+            (not is_nil(section.end_date) and section.end_date >= ^today),
+        select: section,
+        preload: [section_project_publications: [:publication]]
+      )
+    )
+  end
+
+  @doc """
+  Gets all sections that will be affected by forcing the publication update.
+
+  ## Examples
+      iex> get_push_force_affected_sections(project)
+      %{product_count: 1, section_count: 1}
+  """
+  def get_push_force_affected_sections(project_id) do
+    today = DateTime.utc_now()
+
+    Repo.one(
+      from(
+        section in Section,
```

Comment:
This will not return correct counts in all cases.  Force pushing an update will only affect course sections that are exactly on the most recent publication.  This function needs to take that into consideration.  To do this, you will need to pass in (or lookup in the query) the most recent publication id.  Take a look at `Oli.Publishing.fetch_products_and_sections_eligible_for_update/2` to see the logic for how the force publish feature identifies the sections that it can force update.

Reply:
All right! I'll fix it then!